// @nativescript/angular/esm2020/lib/animations/utils.mjs
import { KeyframeAnimation, parseKeyframeDeclarations, animationTimingFunctionConverter } from '@nativescript/core';
const DASH_CASE_REGEXP = /-+([a-z0-9])/g;
export function dashCaseToCamelCase(input) {
    return input.replace(DASH_CASE_REGEXP, (...m) => m[1].toUpperCase());
}
export function createKeyframeAnimation(styles, duration, delay, easing) {
    const info = createKeyframeAnimationInfo(styles, duration, delay, easing);
    return KeyframeAnimation.keyframeAnimationFromInfo(info);
}
const createKeyframeAnimationInfo = (styles, duration, delay, easing) => ({
    isForwards: true,
    duration: duration || 0.01,
    delay,
    curve: getCurve(easing),
    keyframes: styles.map(parseAnimationKeyframe),
});
const getCurve = (value) => animationTimingFunctionConverter(value);
const parseAnimationKeyframe = (styles) => ({
    duration: getKeyframeDuration(styles),
    declarations: getDeclarations(styles),
});
const getKeyframeDuration = (styles) => styles.offset;
function getDeclarations(styles) {
    const unparsedDeclarations = Object.keys(styles).map((property) => ({ property, value: styles[property] }));
    return parseKeyframeDeclarations(unparsedDeclarations);
}
