// @nativescript/angular/esm2020/lib/http/http-client.module.mjs
import { NgModule } from '@angular/core';
import { HttpClientModule, HttpBackend } from '@angular/common/http';
import { NSFileSystem } from '../file-system/ns-file-system';
import { NsHttpBackEnd } from './ns-http-backend';
import * as i0 from "@angular/core";
export class NativeScriptHttpClientModule {
}
NativeScriptHttpClientModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.1.1", ngImport: i0, type: NativeScriptHttpClientModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NativeScriptHttpClientModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.1.1", ngImport: i0, type: NativeScriptHttpClientModule, imports: [HttpClientModule], exports: [HttpClientModule] });
NativeScriptHttpClientModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.1.1", ngImport: i0, type: NativeScriptHttpClientModule, providers: [NSFileSystem, NsHttpBackEnd, { provide: HttpBackend, useExisting: NsHttpBackEnd }], imports: [[HttpClientModule], HttpClientModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.1.1", ngImport: i0, type: NativeScriptHttpClientModule, decorators: [{
            type: NgModule,
            args: [{
                    providers: [NSFileSystem, NsHttpBackEnd, { provide: HttpBackend, useExisting: NsHttpBackEnd }],
                    imports: [HttpClientModule],
                    exports: [HttpClientModule],
                }]
        }] });
